execution/executor-benchmark/src/pipeline.rs

```diff
@@ -152,4 +197,33 @@ where
             handle.join().unwrap()
         }
     }
+
+    pub fn partition_block(
+        block_id: HashValue,
+        partitioner: &ShardedBlockPartitioner,
+        mut transactions: Vec<Transaction>,
+    ) -> ExecutableBlock<Transaction> {
+        let last_txn = transactions.pop().unwrap();
+        assert!(matches!(last_txn, Transaction::StateCheckpoint(_)));
+        let analyzed_transactions = transactions.into_iter().map(|t| t.into()).collect();
+        let mut sub_blocks = partitioner.partition(analyzed_transactions, 2);
+        sub_blocks
+            .last_mut()
+            .unwrap()
+            .sub_blocks
+            .last_mut()
+            .unwrap()
+            .transactions
+            .push(TransactionWithDependencies::new(
+                last_txn,
+                CrossShardDependencies::default(),
+            ));
+        ExecutableBlock::new(block_id, ExecutableTransactions::Sharded(sub_blocks))
+    }
+}
+
+/// Message from partitioning thread to execution thread.
+pub struct ParToExeMsg {
```

Comment:
nit: I would rename this to `ExecuteBlockMsg`

Reply:
fixed